OrbitCore/LinuxEventTracer.cpp

```diff
@@ -98,14 +111,22 @@ void LinuxEventTracer::Run(bool* a_ExitRequested)
         new_events = false;
 
         // Read from all ring buffers, create events and store them in the event_queue.
-        // In order to ensure, no buffer is read constantly while others overflow,
-        // we schedule the reading using round robbin like scheduling.
+        // In order to ensure that no buffer is read constantly while others overflow,
+        // we schedule the reading using round-robin like scheduling.
         for (LinuxPerfRingBuffer& ring_buffer : ring_buffers)
         {
-            uint32_t i = 0;
+            if (*a_ExitRequested) {
```

Comment:
As discussed, this doesn't help to avoid all problems that callstacks are added after exit was requested (which currently causes a warning). If the flag is set directly after this check or the check below, you run into the same issue (or you have to lock access to this variable for the entire block starting from here).

Reply:
This would drastically limit the number of such warnings, but yes, the problem to solve is elsewhere.